docker/auth/auth.py

```diff
@@ -102,12 +102,13 @@ def encode_header(auth):
     return base64.urlsafe_b64encode(auth_json)
 
 
-def parse_auth(entries):
+def parse_auth(entries, fail=False):
```

Comment:
Or considering this is only required for parsing the legacy config, maybe `include_legacy=True` ?